crates/typos-cli/src/file.rs

```diff
@@ -664,6 +773,56 @@ fn fix_buffer(mut buffer: Vec<u8>, typos: impl Iterator<Item = typos::Typo<'stat
     buffer
 }
 
+fn fix_file_name<'a>(
+    path: &std::path::Path,
+    file_name: &'a str,
+    fixes: Vec<typos::Typo<'static>>,
+) -> Result<(), std::io::Error> {
+    let file_name = file_name.to_owned().into_bytes();
+    let new_name = fix_buffer(file_name, fixes.into_iter());
+    let new_name = String::from_utf8(new_name).expect("corrections are valid utf-8");
+    let new_path = path.with_file_name(new_name);
+    std::fs::rename(path, new_path)?;
+    Ok(())
+}
+
+fn select_fix(typo: &typos::Typo<'_>) -> Option<usize> {
```

Comment:
When I run this locally, I get weird output on the right item

![image](https://github.com/user-attachments/assets/61de492d-fb84-4ceb-be6f-972253078cd8)

Reply:
Something still isn't right
![image](https://github.com/user-attachments/assets/3feeb35c-3d4a-4b58-a7b6-410931794172)

And if I ctrl-c,. my cursor no longer shows up

Reply:
I can reproduce the issue with the cursor (but thought something in my setup is broken, apparently it's not my setup), I've really no idea what causes it. I've tried the `inquire` library as well because I thought something is wrong with the `dialoguer` library but that one worked even worse...

Reply:
Oh well, apparently we have to handle Ctrl+C issues ourselves as per https://github.com/console-rs/dialoguer/issues/294.

Reply:
It seems like the available cli libraries for prompts are not really that mature yet.